### PR TITLE
Code Quality: Migrated `CreateDirectoryFromApp` to CsWin32 

### PR DESCRIPTION
**Pull request by gumbarros**
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Part of #14505, #15000 

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes? 

**Additional comments**
Also removed unused methods from a previous PR #15131

### CODE REVIEW

File: src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ internal sealed class DefaultSettingsSerializer : ISettingsSerializer
 
 		public bool CreateFile(string path)
 		{
-			CreateDirectoryFromApp(Path.GetDirectoryName(path), IntPtr.Zero);
+			PInvoke.CreateDirectoryFromApp(Path.GetDirectoryName(path), new SECURITY_ATTRIBUTES());
 
 			var hFile = CreateFileFromApp(path, GENERIC_READ, FILE_SHARE_READ, IntPtr.Zero, OPEN_ALWAYS, (uint)File_Attributes.BackupSemantics, IntPtr.Zero);
 			if (hFile.IsHandleInvalid())
```
Comment (yaira2) [inline src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs:20]: What does `SECURITY_ATTRIBUTES` represent?

Comment (0x5bfa) [reply src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs:20]: Shouldn't that be nint.Zero or something?

Also, in fileapifromapp.h there should be more params.

Can you send me a screenshot of the definition of the function?

Comment (gumbarros) [reply src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs:20]: > What does `SECURITY_ATTRIBUTES` represent?

From the docs:
The **lpSecurityDescriptor** member of the structure specifies a security descriptor for the new directory. If *lpSecurityAttributes* is **NULL**, the directory gets a default security descriptor. The ACLs in the default security descriptor for a directory are inherited from its parent directory. The target file system must support security on files and directories for this parameter to have an effect.

Checking the generated source, if null will do the same thing inside the method.

image


We can just pass null or create the default value like I'm doing , what you think?

Comment (0x5bfa) [reply src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs:20]: Passing null, default or nint.Zero is valid instead.

Comment (gumbarros) [reply src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs]: `nint.Zero` is not valid, passing null instead.

Comment (0x5bfa) [reply src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs]: Thanks!

**Review by 0x5bfa — APPROVED** [review]

**Pull request merged**